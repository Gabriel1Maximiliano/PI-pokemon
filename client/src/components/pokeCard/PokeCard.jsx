import React from 'react'
import { NavLink } from 'react-router-dom';
import imge from '../../images/turma-pokemon-png.png'
import PropTypes from 'prop-types'; 
export const PokeCard = ({name,strength,img,id,type}) => {
  return  (
    <div className='card-image-container' >
        <h2>{name[0].toUpperCase()}{name.substring(1)}</h2>
          <NavLink to={'/home/' +id} >  <img src= {img || imge} className='card-image' alt={`Imagen de ${name}`} /></NavLink> 
        
           <h2>Fuerza</h2>
           <p>{strength}</p>
           <h2>Tipo</h2>
           
            {
              type?.map((type,i)=>(
                <p key={type[i]} >{type[0].toUpperCase()}{type.substring(1)}</p>
              ))
            }
                
    </div>
  )
}
PokeCard.propTypes = {
  name: PropTypes.string.isRequired,
}