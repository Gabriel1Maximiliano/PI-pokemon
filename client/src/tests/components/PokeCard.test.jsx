import { fireEvent, render, screen } from "@testing-library/react";
import { BrowserRouter } from 'react-router-dom';

import { PokeCard } from "../../components/pokeCard/PokeCard";

describe('Pruebas sobre <PokeCard />', () => { 
   
    test('Debe hacer un match con el Snapshot', () => {
       
    const {container} = render(<BrowserRouter><PokeCard name={"Pokeprueba"} strength={35} img={null} id={"78ad527e-0aff-45d9-ac5d-463626452bf0"} type={ [
        "psychic",
        "ice"]} /></BrowserRouter> )
        expect( container ).toMatchSnapshot()
});

test('Debe mostrar el nombre en un h2', () => {
    let testName = "Pokeprueba";
   const { container, getByText } = render(<BrowserRouter><PokeCard name={"Pokeprueba"} strength={35} img={null} id={"78ad527e-0aff-45d9-ac5d-463626452bf0"} type={ [
        "psychic",
        "ice"]} /></BrowserRouter> )

        expect(getByText(testName)).toBeTruthy();

        const h2 = container.querySelector('h2');

        expect(h2.innerHTML).toBe(testName)

});
test('Debe mostrar el ALT de la imagen ', () => {
    let testName = "Pokeprueba";
    render(<BrowserRouter><PokeCard name={"Pokeprueba"} strength={35} img={null} id={"78ad527e-0aff-45d9-ac5d-463626452bf0"} type={ [
        "psychic",
        "ice"]} /></BrowserRouter> );

        expect(screen.getByRole('img').alt).toBe(`Imagen de ${testName}`)
})


})
