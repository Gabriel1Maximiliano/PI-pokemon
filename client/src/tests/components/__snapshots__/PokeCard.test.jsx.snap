// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <PokeCard /> Debe hacer un match con el Snapshot 1`] = `
<div>
  <div
    class="card-image-container"
  >
    <h2>
      P
      okeprueba
    </h2>
    <a
      href="/home/78ad527e-0aff-45d9-ac5d-463626452bf0"
    >
        
      <img
        alt="Imagen de Pokeprueba"
        class="card-image"
        src="turma-pokemon-png.png"
      />
    </a>
    <h2>
      Fuerza
    </h2>
    <p>
      35
    </p>
    <h2>
      Tipo
    </h2>
    <p>
      P
      sychic
    </p>
    <p>
      I
      ce
    </p>
  </div>
</div>
`;
